src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.SolutionChecksumUpdater.cs

```diff
@@ -122,6 +130,50 @@ private static void CancelAndDispose(CancellationTokenSource cancellationSource)
                 // dispose cancellation token source
                 cancellationSource.Dispose();
             }
+
+            private void PushTextChanges(Document oldDocument, Document newDocument)
+            {
+                SourceText oldText = null;
+                SourceText newText = null;
+                if ((oldDocument?.TryGetText(out oldText) == false) ||
+                    (newDocument?.TryGetText(out newText) == false))
```

Comment:
how can old or newDocument be null?

Reply:
if document got deleted or added, it can be null.

Reply:
We get a DocumentChanged event on documents being added/deleted?

Reply:
oh ya. for document changed, document might be always non null. but not 100% sure. not sure we have that guarantee or not.

Reply:
Looking at WorkspaceChangeKind.DocumentChanged, i cannot see how either of these could ever be null.  It's also not clear what that would mean.

Reply:
Code like AbstractObjectBrowserLibraryManager.DocumentChangedAsync would throw if this could ever be null on either side.